imagebitmap: Make ImageBitmap serializable and transferable

tharkum opened the pull request:
According to specification ImageBitmap objects are serializable objects and transferable objects.
https://html.spec.whatwg.org/multipage/#the-imagebitmap-interface:imagebitmap-11

Testing:
 - html/canvas/element/manual/imagebitmap/*
 - html/infrastructure/safe-passing-of-structured-data/*
 - html/webappapis/structured-clone/*
 - workers/semantics/structured-clone/*

tharkum committed:
imagebitmap: Make ImageBitmap serializable and transferable

According to specification ImageBitmap objects are
serializable objects and transferable objects.
https://html.spec.whatwg.org/multipage/#the-imagebitmap-interface:imagebitmap-11

Testing:
 - html/canvas/element/manual/imagebitmap/*
 - html/infrastructure/safe-passing-of-structured-data/*
 - html/webappapis/structured-clone/*
 - workers/semantics/structured-clone/*

Signed-off-by: Andrei Volykhin <andrei.volykhin@gmail.com>

tharkum commented:
@jdm @xiaochengh @mrobinson < Please take a look.

jdm reviewed (approved):
This was easy to review. Thanks!

mrobinson commented:
Removing from the queue in order to clarify if this introduces intermittent test failures. If it does, I think it should wait until intermittency can be resolved. If that requires reviewing and landing #37025, I'm happy to help with that.

mrobinson commented:
If the issues always fail that's fine as long as you justify the new failures. In general, new code should not introduce intermittency as it makes the bots less trustworthy. When possible it's better for tests to fail always than pass only some of the time. 

That said if solving this intermittent behavior is just a matter of landing another PR let's do that first.

tharkum commented:
@mrobinson < Sorry, may be I should just update `*.ini` files for this tests in PR?

tharkum commented:
- Rebased on top of `main` branch
- Added two stable unexpected tests (from https://github.com/servo/servo/actions/runs/15215445669)

@mrobinson < After double check I think I misinterpreted linux WPT results and decided that stable unexpected results are intermittent (which is not true). Please correct me, am I right?

mrobinson commented:
> @mrobinson < After double check I think I misinterpreted linux WPT results and decided that stable unexpected results are intermittent (which is not true).

Thanks for the clarification. That makes perfect sense.
